File: src/main/java/com/owncloud/android/ui/activity/CommunityActivity.java
/*
 * Nextcloud Android client application
 *
 * @author Andy Scherzinger
 * @author Tobias Kaminsky
 * Copyright (C) 2016 Andy Scherzinger
 * Copyright (C) 2016 Nextcloud
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU AFFERO GENERAL PUBLIC LICENSE for more details.
 *
 * You should have received a copy of the GNU Affero General Public
 * License along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.owncloud.android.ui.activity;

import android.os.Bundle;
import android.text.Html;
import android.text.method.LinkMovementMethod;
import android.view.MenuItem;
import android.widget.TextView;

import com.google.android.material.button.MaterialButton;
import com.owncloud.android.R;
import com.owncloud.android.databinding.CommunityLayoutBinding;
import com.owncloud.android.utils.DisplayUtils;
import com.owncloud.android.utils.theme.ThemeButtonUtils;
import com.owncloud.android.utils.theme.ThemeColorUtils;

/**
 * Activity providing information about ways to participate in the app's development.
 */
public class CommunityActivity extends DrawerActivity {

    private CommunityLayoutBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = CommunityLayoutBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        // setup toolbar
        setupToolbar();

        updateActionBarTitleAndHomeButtonByString(getString(R.string.drawer_community));

        // setup drawer
        setupDrawer(R.id.nav_community);

        setupContent();
    }

    private void setupContent() {
        binding.communityReleaseCandidateText.setMovementMethod(LinkMovementMethod.getInstance());

        TextView contributeIrcView = binding.communityContributeIrcText;
        contributeIrcView.setMovementMethod(LinkMovementMethod.getInstance());
        contributeIrcView.setText(Html.fromHtml(getString(R.string.community_contribute_irc_text) + " " +
                                                    getString(R.string.community_contribute_irc_text_link,
                                                              ThemeColorUtils.primaryColorToHexString(this),
                                                              getString(R.string.irc_weblink))));

        TextView contributeForumView = binding.communityContributeForumText;
        contributeForumView.setMovementMethod(LinkMovementMethod.getInstance());
        contributeForumView.setText(Html.fromHtml(getString(R.string.community_contribute_forum_text) + " " +
                                                      getString(R.string.community_contribute_forum_text_link,
                                                                ThemeColorUtils.primaryColorToHexString(this),
                                                                getString(R.string.help_link),
                                                                getString(R.string.community_contribute_forum_forum))));

        TextView contributeTranslationView = binding.communityContributeTranslateText;
        contributeTranslationView.setMovementMethod(LinkMovementMethod.getInstance());
        contributeTranslationView.setText(Html.fromHtml(
            getString(R.string.community_contribute_translate_link,
                      ThemeColorUtils.primaryColorToHexString(this),
                      getString(R.string.translation_link),
                      getString(R.string.community_contribute_translate_translate)) + " " +
                getString(R.string.community_contribute_translate_text)));

        TextView contributeGithubView = binding.communityContributeGithubText;
        contributeGithubView.setMovementMethod(LinkMovementMethod.getInstance());
        contributeGithubView.setText(Html.fromHtml(
            getString(R.string.community_contribute_github_text,
                      getString(R.string.community_contribute_github_text_link,
                                ThemeColorUtils.primaryColorToHexString(this),
                                getString(R.string.contributing_link)))));

        MaterialButton reportButton = binding.communityTestingReport;
        ThemeButtonUtils.colorPrimaryButton(reportButton, this);
        reportButton.setOnClickListener(v -> DisplayUtils.startLinkIntent(this, R.string.report_issue_link));

        binding.communityBetaFdroid.setOnClickListener(
            l -> DisplayUtils.startLinkIntent(this, R.string.fdroid_beta_link));

        binding.communityReleaseCandidateFdroid.setOnClickListener(
            l -> DisplayUtils.startLinkIntent(this, R.string.fdroid_link));

        binding.communityReleaseCandidatePlaystore.setOnClickListener(
            l -> DisplayUtils.startLinkIntent(this, R.string.play_store_register_beta));

        binding.communityBetaApk.setOnClickListener(
            l -> DisplayUtils.startLinkIntent(this, R.string.beta_apk_link));
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        boolean retval = true;
        if (item.getItemId() == android.R.id.home) {
            if (isDrawerOpen()) {
                closeDrawer();
            } else {
                openDrawer();
            }
        } else {
            retval = super.onOptionsItemSelected(item);
        }
        return retval;
    }

    @Override
    protected void onResume() {
        super.onResume();

        setDrawerMenuItemChecked(R.id.nav_community);
    }
}

File: app/src/main/java/com/owncloud/android/ui/activity/CommunityActivity.kt
/*
 * Nextcloud - Android Client
 *
 * SPDX-FileCopyrightText: 2023 Alper Ozturk <alper.ozturk@nextcloud.com>
 * SPDX-FileCopyrightText: 2016 Andy Scherzinger
 * SPDX-FileCopyrightText: 2016 Tobias Kaminsky <tobias@kaminsky.me>
 * SPDX-FileCopyrightText: 2016 Nextcloud
 * SPDX-License-Identifier: AGPL-3.0-or-later OR GPL-2.0-only
 */
package com.owncloud.android.ui.activity

import android.os.Bundle
import android.text.method.LinkMovementMethod
import android.view.MenuItem
import androidx.activity.OnBackPressedCallback
import com.nextcloud.utils.extensions.setHtmlContent
import com.owncloud.android.R
import com.owncloud.android.databinding.CommunityLayoutBinding
import com.owncloud.android.utils.DisplayUtils

/**
 * Activity providing information about ways to participate in the app's development.
 */
open class CommunityActivity : DrawerActivity() {
    lateinit var binding: CommunityLayoutBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = CommunityLayoutBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setupToolbar()
        updateActionBarTitleAndHomeButtonByString(getString(R.string.drawer_community))
        setupDrawer()
        binding.communityReleaseCandidateText.movementMethod = LinkMovementMethod.getInstance()
        setupContributeForumView()
        setupContributeTranslationView()
        setupContributeGithubView()
        setupReportButton()
        setOnClickListeners()
        handleOnBackPressed()
    }

    private fun handleOnBackPressed() {
        onBackPressedDispatcher.addCallback(
            this,
            onBackPressedCallback
        )
    }

    private val onBackPressedCallback = object : OnBackPressedCallback(true) {
        override fun handleOnBackPressed() {
            menuItemId = R.id.nav_all_files
            finish()
        }
    }

    private fun setupContributeForumView() {
        val htmlContent = getString(R.string.community_contribute_forum_text) + " " +
            getString(
                R.string.community_contribute_forum_text_link,
                viewThemeUtils.files
                    .primaryColorToHexString(this),
                getString(R.string.help_link),
                getString(R.string.community_contribute_forum_forum)
            )
        binding.communityContributeForumText.setHtmlContent(htmlContent)
    }

    private fun setupContributeTranslationView() {
        val htmlContent = getString(
            R.string.community_contribute_translate_link,
            viewThemeUtils.files.primaryColorToHexString(this),
            getString(R.string.translation_link),
            getString(R.string.community_contribute_translate_translate)
        ) + " " +
            getString(R.string.community_contribute_translate_text)
        binding.communityContributeTranslateText.setHtmlContent(htmlContent)
    }

    private fun setupContributeGithubView() {
        val htmlContent = getString(
            R.string.community_contribute_github_text,
            getString(
                R.string.community_contribute_github_text_link,
                viewThemeUtils.files.primaryColorToHexString(this),
                getString(R.string.contributing_link)
            )
        )
        binding.communityContributeGithubText.setHtmlContent(htmlContent)
    }

    private fun setupReportButton() {
        val reportButton = binding.communityTestingReport
        viewThemeUtils.material.colorMaterialButtonPrimaryFilled(reportButton)
        reportButton.setOnClickListener {
            DisplayUtils.startLinkIntent(
                this,
                R.string.report_issue_empty_link
            )
        }
    }

    private fun setOnClickListeners() {
        listOf(
            binding.communityBetaFdroid to R.string.fdroid_beta_link,
            binding.communityReleaseCandidateFdroid to R.string.fdroid_link,
            binding.communityReleaseCandidatePlaystore to R.string.play_store_register_beta,
            binding.communityBetaApk to R.string.beta_apk_link
        ).run {
            forEach { pair ->
                pair.first.setOnClickListener {
                    DisplayUtils.startLinkIntent(this@CommunityActivity, pair.second)
                }
            }
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            android.R.id.home -> {
                if (isDrawerOpen) closeDrawer() else openDrawer()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }
}
